### DISCUSSION TITLE
Auto-apply tag to all notes within a list

### DISCUSSION THREAD

**Discussion by DandyLyons** [mgmeyers/obsidian-kanban#231] [Ideas & Feature Requests]
It would be wonderful if there was a way to associate a Kanban column with an Obsidian tag. For example, say I have a Kanban with 3 columns, "Todo", "Working On" and "Finished". I would like every note in the "Working On" column to automatically have the "#status/workingOn" tag applied to it.  And when I drag an item from the "Working On" column to the "Finished" column then the Kanban plugin would automatically remove the "#status/workingOn" tag and apply the "#status/finished" tag. 

This should probably be an optional feature. When you click "Edit list", then under "Mark items in this list as complete" there could be an option called "Apply these tags to items in this list".

**Comment by koenbunders** [comment]
Yes! This please.

**Comment by sjlua** [comment]
New obsidian user here, are tags the categories on the top? Trying this out from notion and the tags I'm used to are like small multiple choice cards that you add under each note.

**Reply by DandyLyons** [reply]
https://help.obsidian.md/tags

**Comment by JosemiGT** [comment]
Is there finally something like this?

**Comment by NeonPunk13** [comment]
I am also super eager to figure something like this out, maybe even help with coding some functionality (I don't suspect my coding ability is up to par with helping properly, _but there's only one way to learn!_) 

I think a function like this will be able to greatly improve workflows. (Combined with Dataview, you could pin a pane that always displays daily todos.  

I was confused on whether or not  this function was possible through the use of metadata, but the documentation on using metadata could be improved for those unfamiliar with Markdown, Obsidian and this plugin in general.

**Comment by waffle-lord** [comment]
In case it helps someone else. For my own use case I managed to just get the tasks and subpath from the kanban file.

Since I just wanted to make some graphs based on the sections. You could get other info from the tasks though.


Here is an example of my pie chart using the kanban sections to count how many quests are in a section... which I wanted to use tags for to make easier, but there doesn't seem to be a way to tag every note in a section of the kanban.

tl;dr - get the tasks and check the subpath to see what section of the kanban it is in
```js
const tasks = dv.page("D&D/Quests").file.tasks.array(); // file.tasks.array() to get the tasks on the page as an array

... 

tasks.forEach(task => {
    ...
    const section = task.section.subpath; // section subpath is the heading for the kanban sections
    ...
});
```

full example using dataviewjs and obsidian charts:

My crappy quest dnd kanban looks like this
![image](https://github.com/user-attachments/assets/75fa8caf-5d74-4dad-8f1a-0dd4513ed161)

chart code:
```js
const tasks = dv.page("D&D/Quests").file.tasks.array(); // file.tasks.array() to get the tasks on the page as an array

const questCounts = new Map();

tasks.forEach(task => {
    const statusCount = questCounts.get(task.section.subpath); // section subpath is the heading for the kanban sections
    
    if (statusCount) {
	    // exists, update it
	    questCounts.set(task.section.subpath, statusCount+1);
	    return;
    }
    
    // status doesn't exist, add it
    questCounts.set(task.section.subpath, 1);
});

const keys = Array.from(questCounts.keys());
const values = Array.from(questCounts.values());

dv.paragraph(`\`\`\`chart
	type: 'pie'
	xTitle: 'Quest Counts'
	labels: [${keys}]
	labelColors: true
	series:
		- data: [${values}]
\`\`\``)
```

The chart:
![image](https://github.com/user-attachments/assets/fc5e31a7-2bc1-44d6-9a03-a3605e1c69ae)


I know this probably isn't how most people want to do this (including me) but I guess it works for now ...
